core/src/html/layout.rs

```diff
@@ -505,15 +534,17 @@ impl<'a, 'gc> LayoutContext<'a, 'gc> {
     /// This function bypasses the text fragmentation necessary for justify to
     /// work and it should only be called internally.
     fn append_text_fragment(&mut self, text: &'a WStr, start: usize, end: usize, span: &TextSpan) {
-        let params = EvalParameters::from_span(span);
-        let text_size = Size::from(self.font.unwrap().measure(text, params, false));
-        let text_bounds = BoxBounds::from_position_and_size(self.cursor, text_size);
-        let mut new_text = LayoutBox::from_text(start, end, self.font.unwrap(), span);
+        if let Some(font) = self.font {
+            let params = EvalParameters::from_span(span);
+            let text_size = Size::from(font.measure(text, params, false));
+            let text_bounds = BoxBounds::from_position_and_size(self.cursor, text_size);
+            let mut new_text = LayoutBox::from_text(start, end, font, span);
 
-        new_text.bounds = text_bounds;
+            new_text.bounds = text_bounds;
 
-        self.cursor += Position::from((text_size.width(), Twips::default()));
-        self.append_box(new_text);
+            self.cursor += Position::from((text_size.width(), Twips::default()));
+            self.append_box(new_text);
+        }
     }
```

Comment:
Sorry if I'm butting in.
Does it make sense for this function to silently do nothing (rather than failing loudly?) if there's no font set?

Reply:
I'm not sure. There's a warning printed in the log when a font isn't found, and this situation here is only when there's no font at all - so someone compiled without the default font. Elsewhere in the file we also just ignore this scenario.

I supposed we could make a fallback-fallback font that's just tofu, but...

Reply:
If there's a warning raised earlier, then that makes sense.

Reply:
> so someone compiled without the default font

If it can only possibly be a programmer bug, I'd say it should still be an unwrap/panic/unreachable/etc.

Reply:
I had a further look into what Flash did here just to make sure I wasn't missing something, and discovered that you can trigger the behavior in Flash with:

```actionscript
var text:TextField = new TextField();
text.text = "test";
text.embedFonts = true; // don't fall back to a device font

var format:TextFormat = new TextFormat(); 
format.size = 14;
format.font = "XXXXX";

text.setTextFormat(format); 

addChild(text);
```

So I think the current behavior would be correct for when no fallback was found, which ActionScript can sometimes force upon us.